azurelinuxagent/common/version.py

```diff
@@ -141,7 +144,10 @@ def is_agent_path(path):
 
 def set_goal_state_agent():
     agent = None
-    pids = [pid for pid in os.listdir('/proc') if pid.isdigit()]
+    if os.path.isdir("/proc"):
+        pids = [pid for pid in os.listdir('/proc') if pid.isdigit()]
+    else:
+        pids = []
```

Comment:
WALinuxAgent-2.2.11 running on openbsd 6.1
    Python: 2.7.13
    Goal state agent: 2.2.11

Could you introduce a new osutil callback that allows OS without `/proc` to implement their own method to get a running process? On OpenBSD, we could implement a callback with the `pgrep` system utility. There are two places that could use it instead of accessing `/proc` directly: `set_goal_state_agent()` in `common/version.py` and `validate_cloud_init()` in `pa/provision/default.py`.